packages/content/types/content.d.ts

```diff
@@ -0,0 +1,18 @@
+import type { Database } from './database';
+import type { QueryBuilder } from './query-builder'
+
+export type contentFunc = (...args: Array<string | Object>) => QueryBuilder;
+
+export interface IContentDocument {
+	dir: string;
+	path: string;
+	extension: '.md' | '.json' | '.yaml' | '.xml' | '.csv' | string;
+	slug: string;
+	createdAt: Date;
+	updatedAt: Date;
+	[key: string]: any;
```

Comment:
I think it would be nicer to use `interface IContentDocument extends Record<string, any>` instead.

Reply:
I agree, I'll update this later